### DISCUSSION TITLE
Upgrade to jersey 2.38 leads to a change of behaviour for the Optional Params

### DISCUSSION THREAD

**Discussion by alainbodiguel** [dropwizard/dropwizard#6533] [Q&A]
There seem to be a change in behaviour about handling empty or null Optional (documented here https://www.dropwizard.io/en/latest/getting-started.html#creating-a-resource-class) when upgrading to Jersey 2.38.
It may be related to https://github.com/eclipse-ee4j/jersey/pull/5194 ?

**Comment by joschi** [comment]
@alainbodiguel Yes, there's a change of behavior and yes, it's related to the GitHub issue you've linked to.

See also https://github.com/eclipse-ee4j/jersey/issues/4790 for a related discussion and https://github.com/dropwizard/dropwizard/pull/4240 for a bit of history.

**Comment by alainbodiguel** [comment]
Thank you.
So does it mean that when Dropwizard upgrades to Jersey 2.38, Optional<String> with a value '' (empty string) will be treated the same as a missing value (param not present, therefore null), i.e. return Optional.empty() instead of Optional.of("")? If yes, is there another way to distinguish these 2 cases?

**Reply by joschi** [reply]
> So does it mean that when Dropwizard upgrades to Jersey 2.38, Optional with a value '' (empty string) will be treated the same as a missing value (param not present, therefore null), i.e. return Optional.empty() instead of Optional.of("")?

Yes, this essentially restores the behavior from before Jersey 2.35.

Unfortunately a _missing_ parameter and an _empty_ parameter are the same for Jersey when using `Optional<T>` and you cannot distinguish between them.

If you need to distinguish between these two cases, you'll have to use the variant without `Optional<T>` and check for `null` vs. empty strings yourself.
